src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorCompilationService.cs

```diff
@@ -53,16 +54,29 @@ public CompilationResult Compile(RelativeFileInfo file)
                 throw new ArgumentNullException(nameof(file));
             }
 
+            var filePath = file.FileInfo.PhysicalPath ?? file.RelativePath;
```

Comment:
Curious: There's something I'm missing in these PRs. Why isn't this enough to get Razor generating pragmas with the full path?